src/dials/algorithms/correlation/plots.py

```diff
@@ -82,6 +82,69 @@ def plot_dims(dims: list, funcs: list) -> dict:
     return d
 
 
+def plot_reachability(nums: np.ndarray, reach: np.ndarray, labels: np.ndarray) -> dict:
+    """
+    Prepares a plotly-style figure of the reachability plot calculated by OPTICS.
+    Args:
+        nums (np.ndarray): array of datasets in OPTICS cluster order - NOT the same as dataset IDS
+        reach (np.ndarray): reachability value between neighbouring datasets
+        labels (np.ndarray): labels for which cluster each point belongs to in OPTICS cluster order
+    Returns:
+        d (dict): plotly-style dictionary of reachability plot
+    """
+
+    unique_labels = set(labels)
+    unique_labels = sorted(unique_labels)
+    n_clusters = max(len(unique_labels) - (1 if -1 in unique_labels else 0), 1)
+
+    import matplotlib
+
+    matplotlib.use("Agg")
+    from matplotlib import pyplot as plt
+
+    colours = plt.cm.nipy_spectral(np.linspace(0.1, 0.9, n_clusters)).tolist()
+
+    if -1 in unique_labels:
+        colours.insert(0, (0, 0, 0, 1))
+    data = []
+    for k, col in zip(unique_labels, colours):
+        isel = np.where(labels == k)[0]
+        data.append(
+            {
+                "x": nums[isel].tolist(),
+                "y": reach[isel].tolist(),
+                "mode": "markers",
+                "type": "bar",
+                "marker": {
+                    "color": "rgb(%f,%f,%f)" % tuple(col[:3]),
+                },
+                "name": "Cluster %i" % k if k >= 0 else "Noise",
+            }
+        )
+
+    d = {
+        "reachability": {
+            "data": data,
+            "layout": {
+                "title": "OPTICS Reachability",
+                "xaxis": {"title": "Cluster-order of Datasets"},
+                "yaxis": {"title": "Reachability Distance Between Datasets"},
+            },
+            "help": """\
+Reachability plot from OPTICS analysis (M. Ankerst et al, 1999, ACM SIGMOD). As this is a density-
```

Comment:
I think it would be good for us to also reference the scikit learn optics page here, so that people can read that to understand more about the parameters?

Reply:
Have added a link to the documentation page